Add config flow to radarr integration

tkdrob opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->
Radarr can now be set up in the UI. Existing yaml configurations are automatically imported and can be safely removed. A breaking change has been made in disabling the Movies count sensor out of caution. This sensor can cause problems with massive databases. If the user still wishes to use it, they may do so.

This now only supports Radarr API v3. Users running versions using older API versions should update their Radarr application before updating Home Assistant.

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add config flow to radarr integration

dependency repo: https://github.com/tkdrob/aiopyarr
This PR depends on #66411

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [x] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue:
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/20550

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [x] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

tkdrob commented:
This PR currently has issues with mypy not detecting attributes from the `coordinator`. I am at a lose on how to fix this one. Whoever reviews this, I expect them to have issues with this change and would love to get some feedback on how to make this better.

farmio commented:
`aiopyarr`, besides being completely type annotated, doesn't seem to expose its types. Consider adding a `py.typed` file to the package. https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-type-hints-for-third-party-library

Other than that you should not use `assert` in production code to unwrap an optional - use for example an `if is (not) None:` instead.

DavidFW1960 commented:
I am running this component as a custom component because the core version doesn't work with API v3. In 2021.12.ob0 I am seeing these errors:
```
2021-12-07 12:10:41 ERROR (MainThread) [homeassistant.config] Platform error: sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config.py", line 887, in async_process_component_config
    platform = p_integration.get_platform(domain)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 530, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 535, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/config/custom_components/radarr/sensor.py", line 10, in <module>
    from homeassistant.const import (
ImportError: cannot import name 'HTTP_OK' from 'homeassistant.const' (/usr/src/homeassistant/homeassistant/const.py)
```

tkdrob commented:
> I am running this component as a custom component because the core version doesn't work with API v3. In 2021.12.ob0 I am seeing these errors:
> 
> ```
> 2021-12-07 12:10:41 ERROR (MainThread) [homeassistant.config] Platform error: sensor
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/config.py", line 887, in async_process_component_config
>     platform = p_integration.get_platform(domain)
>   File "/usr/src/homeassistant/homeassistant/loader.py", line 530, in get_platform
>     cache[full_name] = self._import_platform(platform_name)
>   File "/usr/src/homeassistant/homeassistant/loader.py", line 535, in _import_platform
>     return importlib.import_module(f"{self.pkg_path}.{platform_name}")
>   File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 850, in exec_module
>   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
>   File "/config/custom_components/radarr/sensor.py", line 10, in <module>
>     from homeassistant.const import (
> ImportError: cannot import name 'HTTP_OK' from 'homeassistant.const' (/usr/src/homeassistant/homeassistant/const.py)
> ```

I suggest you try this PR as your custom component. A breaking change was disclosed in the release notes for custom components since the project does not maintain those.

DavidFW1960 commented:
I was just looking at that. Do I just need to grab the files from the radarr folder of your commit to try it?

DavidFW1960 commented:
looks like I need the config_flow as well?

tkdrob commented:
> I was just looking at that. Do I just need to grab the files from the radarr folder of your commit to try it?

Yes but you also need to add two things. Version in the manifest, can be anything you want. And this decorator directly above the `RadarrConfigFlow` class:
```
@config_entries.HANDLERS.register(DOMAIN)
```

DavidFW1960 commented:
ok so line 55 & 56 of config_flow.py
```
@config_entries.HANDLERS.register(DOMAIN)
class RadarrConfigFlow(ConfigFlow, domain=DOMAIN):
```
and add line 10 as
```
 import from homeassistant import config_entries
```
and add version in manifest like:
```
  "domain": "radarr",
  "name": "Radarr",
  "documentation": "https://www.home-assistant.io/integrations/radarr",
  "requirements": ["aiopyarr==21.12.1"],
  "codeowners": ["@tkdrob"],
  "config_flow": true,
  "iot_class": "local_polling",
  "version": "1.1.0"
```
then save as a custom component?

DavidFW1960 commented:
so I get this error:
```
2021-12-08 10:25:50 ERROR (MainThread) [homeassistant.config] Platform error: sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config.py", line 887, in async_process_component_config
    platform = p_integration.get_platform(domain)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 530, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 535, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/config/custom_components/radarr/sensor.py", line 8, in <module>
    from homeassistant.components.radarr import RadarrEntity
ImportError: cannot import name 'RadarrEntity' from 'homeassistant.components.radarr' (/usr/src/homeassistant/homeassistant/components/radarr/__init__.py)
```
I made the changes in  config_flow.py as I said in the above post. should I have made them in sensor.py instead?

tkdrob commented:
> so I get this error:
> 
> ```
> 2021-12-08 10:25:50 ERROR (MainThread) [homeassistant.config] Platform error: sensor
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/config.py", line 887, in async_process_component_config
>     platform = p_integration.get_platform(domain)
>   File "/usr/src/homeassistant/homeassistant/loader.py", line 530, in get_platform
>     cache[full_name] = self._import_platform(platform_name)
>   File "/usr/src/homeassistant/homeassistant/loader.py", line 535, in _import_platform
>     return importlib.import_module(f"{self.pkg_path}.{platform_name}")
>   File "/usr/local/lib/python3.9/importlib/__init__.py", line 127, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 850, in exec_module
>   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
>   File "/config/custom_components/radarr/sensor.py", line 8, in <module>
>     from homeassistant.components.radarr import RadarrEntity
> ImportError: cannot import name 'RadarrEntity' from 'homeassistant.components.radarr' (/usr/src/homeassistant/homeassistant/components/radarr/__init__.py)
> ```
> 
> I made the changes in config_flow.py as I said in the above post. should I have made them in sensor.py instead?

Please see latest commit. Also keep any further support discussion in the issue you created. Thanks

frenck reviewed (changes requested):
So... here is the deal. I started reviewing, but... this PR is just too big.

There are many different things happening here, that really do not need to be in the same PR.

For example, adding a new library really doesn't have to be in the same PR (probably should be done before this PR), moving stuff around (like the const file), the coordinator, the config flow. Even things like tests (besides the config flow, or limited to the changeset).

Smaller make things go quicker through the pipelines. But a +1,647 −585 PR, is really hard for any reviewer.

I suggest to split to up before going futher.

thecode committed:
Fix webostv notify service (#66760)

teharris1 committed:
Bump pyinsteon to 1.0.16 (#66759)

thecode committed:
Handle default notify data in webostv (#66770)

balloob committed:
Bump aiohue to 4.2.1 (#66823)

balloob committed:
Bumped version to 2022.2.9

balloob committed:
Bump hass-nabucasa to 0.53.0 (#66826)

madmat777 committed:
Add broadlink lb2 support (#63530)

* Update const.py

* Update to support LB2 version smart bulbs in Broadlink integration

* Update const.py

Added Space.

* Update updater.py

Updated so that LB2 lights use the LB1 update manager.

frenck committed:
Downgrade log warning->info for unregistered webhook message (#66830)

balloob committed:
Merge pull request #66829 from home-assistant/rc

gjohansson-ST committed:
Improve code quality workday (#66446)

* Code quality workday

* Modify from review

* Modify from review 2

* Fix mypy

jjlawren committed:
Initialize AlarmDecoder binary_sensor state as off instead of unknown (#65926)

epenet committed:
Add installed apps to samsungtv sources (#66752)

Co-authored-by: epenet <epenet@users.noreply.github.com>

jjlawren committed:
Bump SoCo to 0.26.3 (#66834)

bvweerd committed:
Fix eq3btsmart setting HVAC modes (#66394)

* Partly reverse preset incompatibility

It seems like some presets are unsupported by the native climate control of Home Assistant core. This change reverts the previous preset changes causing issues. It worked perfect with simple-thermostat custom lovelace card.

* Remove priority of preset above HVAC mode 

If a preset was available of the given command, the hvac mode change was ignored. This can result in HVAC settings are ignored. By removing the check for a preset, the preset does not supersede the HVAC mode anymore

* Revert "Partly reverse preset incompatibility"

This reverts commit 10fdc8eef457c369a042c631376ed33f29d0d8bf.

emontnemery committed:
Add support for validating and serializing selectors (#66565)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

cdce8p committed:
Add type ignore error codes [auth] (#66774)

kbickar committed:
Add SleepIQ device type (#66833)

Co-authored-by: J. Nick Koston <nick@koston.org>

balloob committed:
Bump hass-nabucasa to 0.53.1 (#66845)

bdraco committed:
Startup with an emergency self signed cert if the ssl certificate cannot be loaded (#66707)

balloob committed:
Validate in split_entity_id (#66835)

Commit:
[ci skip] Translation update

raman325 committed:
Create zwave_js ping button at the right time (#66848)

* Create zwave_js ping button at the right time

* fix tests

DurgNomis-drol committed:
Code enhancements for ISS (#66813)

* Code enhancements for ISS

* Assert options

ludeeus committed:
Add missing hass argument in async_request_config call (#66864)

cdce8p committed:
Don't run pytest CI jobs on push to forks (#66870)

gregsheremeta committed:
typo fix networrk --> network (#66878)

AnilDaoud committed:
Kaiterra type issue (#66867)

cdce8p committed:
Fix braviatv typing (#66871)

cdce8p committed:
Fix modbus typing (#66872)

bdraco committed:
Ensure dhcp can still discover new devices from device trackers (#66822)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

cdce8p committed:
Fix typo [recorder] (#66879)

janiversen committed:
Add support for INT8 and UINT8 in Modbus (#66889)

cdce8p committed:
Use assignment expressions [K-Z] (#66881)

cdce8p committed:
Use assignment expressions [A-I] (#66880)

cdce8p committed:
Use assignment expressions [other] (#66882)

bdraco committed:
Fix yeelight config flow ip update and timeout (#66883)

thecode committed:
Add Shelly gen2 error sensors (#66825)

klaasnicolaas committed:
Add Pure Energie integration (#66846)

trekie86 committed:
Update srpenergy dependency to 1.3.6 (#66821)

kbickar committed:
Create button entities for SleepIQ (#66849)

gjohansson-ST committed:
Bump pysensibo to 1.0.4 (#66886)

jingsno committed:
Fix Mill Gen1 Climate Control (#66899)

Fixes Mill Gen1 Climate Control, so it correctly returns the current status of the heating element.

rytilahti committed:
Enable consumable sensors per default for xiaomi_miio vacuums (#66843)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

jjlawren committed:
Proxy Plex thumbnail images in media browser (#66702)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Bump aiodiscover to 1.4.8 (#66892)

jt-nti committed:
Update RSS feed template (#62966)

rklomp committed:
Prefix sma sensor name (#65234)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

gjohansson-ST committed:
Bump pysensibo to v1.0.5 (#66906)

austinmroczek committed:
bump total_connect_client to 2022.2.1 (#66907)

ctalkington committed:
Update rokuecp to 0.14.1 (#66894)

Commit:
[ci skip] Translation update

bdraco committed:
Add support for WiZ diagnostics (#66817)

allenporter committed:
Update nest climate set_temperature to allow hvac_mode (#66909)

janiversen committed:
Remove async_setup_component() from tests (#66905)

chishm committed:
Add media browser support to dlna_dmr (#66425)

chemelli74 committed:
Fritz: fix unbound topology (#66877)

Co-authored-by: Michael <35783820+mib1185@users.noreply.github.com>
Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

emontnemery committed:
Deduplicate code in cast media_player (#66815)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

raman325 committed:
Improve zwave_js notification event handling (#66790)

balloob committed:
Bump frontend to 20220220.0 (#66919)

ludeeus committed:
Bump aiogithubapi from 22.2.0 to 22.2.3 (#66924)

oischinger committed:
Add service configuration URL for vicare (#66927)

frenck committed:
Introduce const file in LaMetric (#66929)

raman325 committed:
Add zwave_js.event automation trigger (#62828)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

janiversen committed:
Set slave default to 0, as already documented in Modbus (#66921)

ludeeus committed:
Use GraphQL for GitHub integration (#66928)

ludeeus committed:
Enable all GitHub sensors by default (#66931)

gjohansson-ST committed:
Bump pysensibo to v1.0.6 (#66930)

ludeeus committed:
Add Latest Tag sensor to GitHub integration (#66932)

timmo001 committed:
Add discussions sensors to GitHub (#66937)

chishm committed:
Avoid accessing hass.data in test_play_media_didl_metadata (#66939)

thecode committed:
Bump aiowebostv to 0.1.3 (#66942)

dewdropawoo committed:
Fix broken aftership sensor after pyaftership 21.11.0 bump (#66855)

* Fix update to pyaftership 21.11.0

Variable name collision and a missing property access was causing this sensor to always return zero elements.

* Move subscripting out of awaited statement

klaasnicolaas committed:
Update Pure Energie integration (#66946)

* Remove service entry_type

* Set raise on progress

rikroe committed:
Bump bimmer_connected to 0.8.11 (#66951)

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

CoMPaTech committed:
Plugwise bump module version to fix heating-state and OnOff devices (#66936)

Commit:
[ci skip] Translation update

davet2001 committed:
Use new enums in vizio tests (#62710)

* Use new enums in vizio tests

* Code review: revert wrong conftest changes

* Code review: Revert incorrect removal of vizio const

* Reinstate wrongly reverted files

* Fix double line

* Fix new test after rebase

epenet committed:
Use hass.add_job in samsungtv (#66976)

Co-authored-by: epenet <epenet@users.noreply.github.com>

julesxxl committed:
Fix typo in const.py (#66856)

emontnemery committed:
Add WS API for removing a config entry from a device (#66188)

* Add WS API for removing a config entry from a device

* Address review comments

* Address review comments

* Remove entity cleanup from ConfigEntries

* Update + add tests

* Improve comments in test

* Add negative test

* Refactor according to review comments

* Add back async_remove_config_entry_device

* Remove unnecessary error handling

* Simplify error handling

epenet committed:
Bump samsungtvws to 1.7.0 (#66978)

Co-authored-by: epenet <epenet@users.noreply.github.com>

G-Two committed:
Bump to subarulink 0.4.2 (#66403)

frenck committed:
Merge branch 'master' into dev

jkeljo committed:
Bump greeneye_monitor to v3.0.3 (#66973)

starkillerOG committed:
Motion blinds add VerticalBlindLeft support (#66961)

bdraco committed:
Add support for dual head WiZ devices (#66955)

cdce8p committed:
Remove `setup.py` (#66023)

Kirmas committed:
Add aditional sensors for dmaker.airfresh.a1/t2017 to xiaomi_miio (#66370)

uvjustin committed:
Allow stream log level to change at runtime (#66153)

bdraco committed:
Add additional WiZ OUIs (#66991)

mib1185 committed:
Enable sensors based on wan scenario in Fritz!Tools (#66944)

frenck committed:
Add Radio Browser integration (#66950)

DeerMaximum committed:
Fix nina warnings in city states (#65914)

avee87 committed:
Rename manual alarm integrations (#66979)

mdegat01 committed:
Support variables in templates with timeout (#66990)

allenporter committed:
Simplify nest placeholder image loading and share across all cameras (#66580)

rytilahti committed:
Create LED switches for tplink dimmers (#66839)

bdraco committed:
Add support for auto target fan state in HomeKit fans (#66383)

tkdrob committed:
Refactor tests for modem_callerid (#59691)

* Refactor tests for modem_callerid

* uno mas

* uno mas

* uno mas

javicalle committed:
Restore states for RFLink binary_sensors (#65716)

tkdrob committed:
Deprecate "wanted" sensor in radarr (#63818)

* Remove invalid "wanted" sensor from radarr

* uno mas

dgomes committed:
Cast string back to datetime in Sensor Filter (#65396)

balloob committed:
Allow deleting files from media source (#66975)

janiversen committed:
Correct switch verify to handle discret_read in Modbus (#66890)

janiversen committed:
Allow multiread in modbus binary_sensor (#59886)

bdraco committed:
Only set require_restart on config entry reload if its not recoverable (#66994)

iMicknl committed:
Fix binary sensor translations for carbon_monoxide (#66891)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

bdraco committed:
Ensure WiZ can still setup with old firmwares (#66968)

frenck committed:
Extend Plugwise DeviceInfo (#66619)

jjlawren committed:
Enable fallback polling for Sonos microphone binary_sensor (#66299)

iMicknl committed:
Remember user and hub after input in ConfigFlow (#66608)

dgomes committed:
Add unique_id to the filter component  (#65010)

* Optional manually defined uniqueid

* move to _attr

epenet committed:
Add door and lock status to Renault integration (#66698)

* Add coordinator for lock status

* Add fixture for lock status

* Add lock status binary sensor

* Add to test constants

* Adjust conftest

* Fix inverted state

* Add door status

Co-authored-by: epenet <epenet@users.noreply.github.com>

farmio committed:
add apparent and reactive power DeviceClass (#65938)

gjohansson-ST committed:
Implement number platform for Sensibo (#66898)

iMicknl committed:
Update pyoverkiz to 1.3.6 (#66997)

thecode committed:
Fix Shelly event handling (#67000)

kbickar committed:
Add Multi factor authentication support for Sense (#66498)

Co-authored-by: J. Nick Koston <nick@koston.org>

corneyl committed:
Add sensors for next Picnic deliveries (#66474)

janiversen committed:
Diferentiate between attr_name and entity_id in Modbus tests (#66999)

rubenverhoef committed:
Additional MQTT light command templates (#63361)

Co-authored-by: jbouwh <jan@jbsoft.nl>

chishm committed:
Add dlna_dms integration to support DLNA Digital Media Servers (#66437)

slovdahl committed:
Improve Vallox filter remaining time sensor (#66763)

GitHub-Action committed:
[ci skip] Translation update

frenck committed:
Use selectors in Open-Meteo configuration flow (#67004)

mdegat01 committed:
Fix MQTT lights tests using `STATE_OFF` (#67011)

bdraco committed:
Add diagnostics support to flux_led (#67012)

eifinger committed:
Cleanup_google_travel_time_tests (#66868)

eifinger committed:
Waze travel time sensor tests (#66558)

eifinger committed:
google_travel_time: always resolve zones (#66165)

eifinger committed:
waze_travel_time: always resolve zones (#66162)

balloob committed:
Allow supported brands in manifests (#67015)

raman325 committed:
Bump zwave-js-server-python to 0.35.1 (#67014)

balloob committed:
Fix radio browser on Sonos (#67017)

kbickar committed:
Add Switch entity to SleepIQ (#66966)

Co-authored-by: J. Nick Koston <nick@koston.org>

epenet committed:
Bump renault-api to 0.1.9 (#67016)

Co-authored-by: epenet <epenet@users.noreply.github.com>

bdraco committed:
Add configuration_url to lookin (#67021)

bdraco committed:
Add support for climate fan and oscillate mode to HomeKit (#66463)

emontnemery committed:
Add Google Cast groups to device registry (#66805)

jjlawren committed:
Add log message when Plex library section not found (#66820)

scop committed:
Add service info for upcloud entities (#61740)

frenck committed:
Fix nightly builder (#67022)

Sjoerdfc committed:
Add the ICAO 24-bit address to the OpenSky sensor events (#66114)

raman325 committed:
Log error when using zwave_js 'refresh_value' on ping button/node status sensor (#66847)

* Raise when using 'zwave_js.refresh_value' on ping button

* Revert test change

* block till done

* Switch from raising an exception to logging an error for both the ping button and node status sensor

* missed commit

pascalwinters committed:
Bump PySwitchbot to 0.13.3 (#67025)

epenet committed:
Cleanup Renault tests (#67030)

Co-authored-by: epenet <epenet@users.noreply.github.com>

gjohansson-ST committed:
Bump pysensibo to v1.0.7 (#67032)

cdce8p committed:
Cleanup after setup.py removal (#67036)

epenet committed:
Add Dacia as supported brand to Renault (#67029)

Co-authored-by: epenet <epenet@users.noreply.github.com>

cdce8p committed:
Upgrade pwmled to 1.6.10 (#67034)

ludeeus committed:
Deprecate the updater integration (#67038)

gjohansson-ST committed:
Improve code quality moon (#66461)

* Code quality moon

* Fix review comments

gjohansson-ST committed:
Add mac address as connection for Sensibo devices (#67035)

MartinHjelmare committed:
Bump pyicloud to 1.0.0 (#67037)

eifinger committed:
Use length_util conversion (#67049)

chemelli74 committed:
Bump pyuptimerobot to 22.2.0 (#67041)

tkdrob commented:
@frenck Can you change the target of this PR to: `tkdrob-radarr-config-flow`?

ctalkington committed:
Use aiopyarr for sonarr (#65349)

tkdrob committed:
Merge branch 'tkdrob-radarr-config-flow' of https://github.com/home-assistant/core into radarr_config_flow

tetienne committed:
Add low speed Overkiz cover (#66750)

epenet committed:
Expose Samsung wrapper as async (#67042)

Co-authored-by: epenet <epenet@users.noreply.github.com>

iMicknl committed:
Improve code quality for Overkiz integration (#67060)

thecode committed:
Bump aioshelly to 1.0.10 (#67056)

bdraco committed:
Add newly discovered samsungtv OUI (#67059)

balloob committed:
Re-org device automations (#67064)

Co-authored-by: Franck Nijhof <git@frenck.dev>

balloob committed:
Add a new validate config WS command (#67057)

balloob committed:
Block peer certs on supervisor (#66837)

Co-authored-by: Pascal Vizeli <pvizeli@syshack.ch>
Co-authored-by: Mike Degatano <michael.degatano@gmail.com>

balloob committed:
Add source name to radio browser media source (#67077)

balloob committed:
Bump frontend to 20220222.0 (#67078)

mib1185 committed:
Fix naming of device entities created by Fritz!Tools (#67076)

marcelveldt committed:
Add (basic) diagnostics support for Hue integration (#67074)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

iMicknl committed:
Bump PyOverkiz and improve code quality (late review) (#67075)

chemelli74 committed:
Fix missing uptime sensor in some Fritz scenarios (#67073)

* Fix missing uptime sensor in some Fritz scenarios

* apply review comment

GitHub-Action committed:
[ci skip] Translation update

bdraco committed:
Add diagnostics support to lutron_caseta (#67079)

jjlawren committed:
Add Plex scan_clients button, enable autoscan (#67055)

Co-authored-by: Robert Svensson <Kane610@users.noreply.github.com>

mib1185 committed:
Improve tests of Fritz!Tools (part1) (#66972)

chemelli74 committed:
Fritz device_trackers for non mesh devices (#67006)

balloob committed:
Improved local media ID handling (#67083)

cdce8p committed:
Fix type issues [litterrobot] (#67092)

cdce8p committed:
Fix type issues [firmata] (#67093)

MartinHjelmare committed:
Use RequestError in tradfri (#67101)

emontnemery committed:
Deprecate manual MQTT configuration available in config flow (#66247)

balloob committed:
Track hidden items in media source (#67096)

cdce8p committed:
Fix type issues [geniushub] (#67095)

PoltoS committed:
Add suggested area to the Z-Wave.Me integration (#66986)

* Add Z-Wave.Me Device Info

* fix

* fix

* small fix

Co-authored-by: Dmitry Vlasov <kerbalspacema@gmail.com>

cdce8p committed:
Remove unused attribute [litterrobot] (#67106)

cdce8p committed:
Fix type issues [fireservicerota] (#67094)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

rfleming71 committed:
Add tools to octoprint when the printer comes back online (#59666)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

MatthiasLohr committed:
Add number platform to tolo integration (#66799)

MatthiasLohr committed:
Add timer sensors for TOLO (#66938)

bdraco committed:
Enable strict typing for powerwall (#65577)

Kisty committed:
Allow multidict 6.0.2+ to fix ZHA, gTTS and other integrations (#67046)

jbouwh committed:
Mqtt improve test coverage (#66279)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

jbouwh committed:
Add MQTT publish ACK timeout test (#67062)

frenck committed:
Revert "Block peer certs on supervisor" (#67104)

Kane610 committed:
Make type checking pass for deCONZ init, gateway and services (#66054)

* Type and enable type checking for init, config_flow, diagnostics, gateway and services

* Fix import

* Fix review comment

ludeeus committed:
Bump awesomeversion from 22.1.0 to 22.2.0 (#67107)

ludeeus committed:
Bump pyhaversion from 21.11.1 to 22.02.0 (#67108)

dmulcahey committed:
Bump ZHA quirks to 0.0.67 (#67109)

cdce8p committed:
Improve sonos ConfigFlow registration (#67110)

tkdrob committed:
Deprecate yaml config for fritzbox callmonitor (#61762)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

tkdrob committed:
Add Phone Modem call reject button (#66742)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

epenet committed:
Add Remote Engine Start status to Renault integration (#67028)

Co-authored-by: epenet <epenet@users.noreply.github.com>

ludeeus committed:
Add homeassistant to partial backup service (#67117)

cdce8p committed:
Fix type issues [mobile_app] (#67091)

jjlawren committed:
Refactor Sonos media metadata handling (#66840)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

EiNSTeiN- committed:
Add Humidifier support to zwave_js (#65847)

patagonaa committed:
Correctly handle missing mpd albumart (#66771)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bdraco committed:
Remove effects from WiZ wall dimmer switches (#67097)

balloob committed:
Mobile app: Drop descriptive emoji name support (#67120)

epenet committed:
Import tag (#64539)

Co-authored-by: epenet <epenet@users.noreply.github.com>

eifinger committed:
Cleanup Waze_travel_time_sensor_tests (#67047)

jkeljo committed:
Create greeneye_monitor entities when monitor connects (#66710)

milanmeu committed:
Add Nanoleaf Swipe Device Trigger (#66195)

iMicknl committed:
Add support for Atlantic Electrical Heater in Overkiz integration (#67045)

Co-authored-by: J. Nick Koston <nick@koston.org>

sophof committed:
Fix derivative integration showing unexpected spikes (#65528)

zvldz committed:
Add telegram message_tag, disable_notification, parse_mode (#63604)

Co-authored-by: root <zvldz@users.noreply.github.com>

R0nd committed:
Support setting volume in lg_netcast media_player (#58126)

Co-authored-by: Artem Draft <Drafteed@users.noreply.github.com>

flabbamann committed:
Allow sending telegram stickers from sticker packs (#57482)

DeerMaximum committed:
Fix missing nina start value (#66869)

iMicknl committed:
Fix RTS device delays in Overkiz integration (#67124)

MartinHjelmare committed:
Add mysensors remove device support (#67128)

elupus committed:
Use opt in device removal for rfxtrx (#58252)

raman325 committed:
Add zwave_js light support for HSM-200 V1 (#67089)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

emontnemery committed:
Improve Tasmota device removal (#66811)

Kane610 committed:
Tweak UniFi client tracker (#67129)

cdce8p committed:
Improve typing [util.decorator] (#67087)

bramkragten committed:
Update frontend to 20220223.0 (#67130)

frenck committed:
Bump version to 2022.4.0dev0 (#67132)

tkdrob committed:
Merge branch 'dev' of https://github.com/home-assistant/core into radarr_config_flow

jjlawren committed:
Fix Sonos radio metadata processing with missing data (#67141)

GitHub-Action committed:
[ci skip] Translation update

balloob committed:
Fix SQL sensor (#67144)

balloob committed:
Media source to verify domain to avoid KeyError (#67137)

soluga committed:
Don't try to resolve state if native_value is Null (#67134)

balloob committed:
Allow get_states to recover (#67146)

tkdrob committed:
Bump aiopyarr to 22.2.2 (#67149)

kbickar committed:
SleepIQ Dependency update (#67154)

gagebenne committed:
Bump pydexcom to 0.2.3 (#67152)

bdraco committed:
Use compact encoding for JSON websocket messages (#67148)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

frenck committed:
Remove deprecated YAML configuration from PVOutput (#67162)

* Remove deprecated YAML configuration from PVOutput

* Clean up platform schema

frenck committed:
Remove deprecated YAML configuration from Whois (#67163)

* Remove deprecated YAML configuration from Whois

* Clean up platform schema

frenck committed:
Remove deprecated APNS integration (#67158)

frenck committed:
Remove deprecated YAML configuration from CPU Speed (#67166)

rikroe committed:
Remove deprecated yaml config from bmw_connected_drive (#66965)

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

janiversen committed:
Allow "slave" parameter in modbus service calls (#66874)

* Allow "slave" parameter in modbus service calls.

frenck committed:
Remove deprecated Time of Flight integration (#67167)

rforro committed:
Presets for single ZONNSMART TRV (#67157)

* Presets for single ZONNSMART TRV

* added zonnsmart climate tests

* black8 fix

tkdrob committed:
Merge branch 'dev' of https://github.com/home-assistant/core into radarr_config_flow

tkdrob committed:
update

homeassistant commented:
Hi @GitHub-Action, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#60847).

Once you do that we will be able to review and accept this pull request.

Thanks!

tkdrob commented:
#67201
